frontend/src/Context/PreviewPublisherProvider/usePreviewPublisher/usePreviewPublisher.tsx

```diff
@@ -228,10 +235,19 @@ const usePreviewPublisher = (): PreviewPublisherContextType => {
           }
         : undefined;
 
+    if (!isDeviceStoreReady) {
+      await deviceStoreRef.current.init();
+      videoSource = deviceStoreRef.current.getConnectedDeviceId('videoinput');
+      audioSource = deviceStoreRef.current.getConnectedDeviceId('audioinput');
+      setIsDeviceStoreReady(true);
```

Comment:
I think you could probably extract all of this stuff away into a `useStoredDevices()` (or similar) hook  that handles this and just returns a couple variables of state in an object like maybe:
```js
const {deviceStoreReady, storedAudioDevice, storedVideoDevice} = useStoredDevices();
```

Reply:
hmm if that's okay I'd like to keep it as-is, at least for now. I tried moving it all in its own hook but I'm not able to get it to work 😅 I tried variety of things but yea no luck